src/sentry/snuba/metrics/naming_layer/mapping.py

```diff
@@ -99,20 +99,17 @@ def is_private_mri(internal_name: TransactionMRI | SessionMRI | str) -> bool:
 
 
 def _extract_name_from_custom_metric_mri(mri: str) -> str | None:
-    match = MRI_SCHEMA_REGEX.match(mri)
-    if match is None:
+    parsed_mri = parse_mri_lenient(mri)
```

Comment:
Took the opportunity to remove all of our dependencies from the regex, which now should not be used directly.